### ISSUE TITLE
`no_leading_underscores_for_local_identifiers` still returns false positive in Dart 2.16

### ISSUE THREAD

**Issue by Pante** [dart-lang/linter#3216]
**Describe the issue**
The following code snippet produces a false positive due to `no_leading_underscores_for_local_identifiers` on Dart 2.16. 
```dart
class Foo {

  int? _bar;

  Foo([this._bar]);

}
```

It seems to have been fixed in #3127 which was subsequently included in `1.18.0` and Dart 2.16's change-log states that it bundles linter `1.18.0`. We've tested it in a clean repository and a clean project & installation but it still fails.

**Comment by mockturtl** [comment]
Can confirm.  

```dart
class Bar {
  final int _x;
  Bar([this._x = 0]);  // Avoid leading underscores for local identifiers. • no_leading_underscores_for_local_identifiers
}
```

Related: I also get false positives in a `catch` clause (fixed in https://github.com/dart-lang/linter/issues/3126).  These are new since Dart 2.15.  

```dart
try {} on Exception catch (e, _) {}  // Avoid leading underscores for local identifiers. • no_leading_underscores_for_local_identifiers
```

(Granted, the easy workaround is to omit the unused stacktrace.)

```sh
$ dart --version
Dart SDK version: 2.16.0 (stable) (Mon Jan 31 15:28:59 2022 +0100) on "linux_x64"
```

**Comment by Pante** [comment]
I'm closing this issue as it seems to be fixed and it no longer affects us.

**Comment by pq** [comment]
Great.  Thanks for closing the loop!

**Comment by Technorocker** [comment]
Not sure if I should make a new issue although it certainly fits this thread. I just finally upgraded my flutter to 3.3.0 from 2.10 and with the newest versions on everything my analyzer now shows like 200 issues. 95% of those are Avoid leading underscores for local identifiers. I am using Dart 2.18 and in my pubspec file my linter is `flutter_lints: ^2.0.1`

It has classified every underscore as this warning.

![dart linter error](https://user-images.githubusercontent.com/35610201/188321597-3073b15d-9e7b-4d3e-961e-a1d92c68dfea.png)

**Comment by pq** [comment]
@Technorocker: could you share some of the places that are getting wrongly reported?  Thanks!

**Comment by Technorocker** [comment]
My apologies, I just removed the suppression of that analysis option to show you and as I took a closer look at them and started going through each one I did realize that there were just that many times I used the underscore... apparently too much lol. I'm down to 80 "Problems" and indeed they are all local identifiers so far.

**Comment by pq** [comment]
No worries!

Thanks for following up. 👍
